tests/functional/test_single_model_vehicle.py

```diff
@@ -51,23 +61,150 @@ def test_run_inference(self, vehicle_adas_model_downloader,
-        out_name = 'softmax_tensor'
-        output = infer(imgs_v1_224, input_tensor=in_name, grpc_stub=stub,
-                       model_spec_name='resnet',
+        imgs_v1_384 = np.ones((1, 3, 384, 672))
+        in_name = 'data'
+        out_name = 'detection_out'
+        output = infer(imgs_v1_384, input_tensor=in_name, grpc_stub=stub,
+                       model_spec_name='vehicle-detection',
                        model_spec_version=None,
                        output_tensors=[out_name])
         print("output shape", output[out_name].shape)
-        assert output[out_name].shape == (1, 1001), ERROR_SHAPE
+        assert output[out_name].shape == (1, 1, 200, 7), ERROR_SHAPE
+
+
+    def test_run_inference_img(self, vehicle_adas_model_downloader,
+                           vehicle_adas_data_downloader,
+                           start_server_single_vehicle_model,
+                           create_grpc_channel):
+        """
+        <b>Description</b>
+        Submit request to gRPC interface serving a single vehicle model
+
+        <b>input data</b>
+        - directory with the model in IR format
+        - docker image with ie-serving-py service
+
+        <b>fixtures used</b>
+        - model downloader
+        - service launching
+
+        <b>Expected results</b>
+        - response contains proper numpy shape
+
+        """
+
+        _, ports = start_server_single_vehicle_model
+        imgs_path =  os.path.join(vehicle_adas_data_downloader, "data", "annotation_val_images")
+
+        img_files = os.listdir(imgs_path)
+        imgs = np.zeros((0,3,384,672), np.dtype('<f'))
+        input_img = self.load_image(os.path.join(imgs_path,"image_000015.jpg"), 672, 384)
+        imgs = np.append(imgs, input_img, axis=0)
+        
+        batch_size = 1
+        # Connect to grpc service
+        stub = create_grpc_channel('localhost:{}'.format(ports["grpc_port"]),
+                                   PREDICTION_SERVICE)
+
+        in_name = 'data'
+        out_name = 'detection_out'
+        output = infer(imgs, input_tensor=in_name, grpc_stub=stub,
+                       model_spec_name='vehicle-detection',
+                       model_spec_version=None,
+                       output_tensors=[out_name])
+        print("output shape", output[out_name].shape)
+        assert output[out_name].shape == (1, 1, 200, 7), ERROR_SHAPE
+        
+        detections_sum = 0
+        result = output[out_name]
+        print("result:" + str(result))
+        for x in range(0, imgs.shape[0] - batch_size + 1, batch_size):
+            img = imgs[x:(x + batch_size)]       
+            print("img: " + str(img))
+            for y in range(0,img.shape[0]):
+                for i in range(0, 200*batch_size-1):
+                    detection = result[:,:,i,:]
+                    print("detection: " + str(detection))
+                    if detection[0,0,2] > 0.5 and int(detection[0,0,0]) == y:
+                        detections_sum+=1
```

Comment:
We do inference on only one image and the batch size is hard-coded to 1. 
Could you simplify this block? I think we only need one loop here. Could you check something like 
this:
```
for i in range(0, 200):
    detection = result[0,0,i]
    if detection[2] > 0.5:
        detections_sum+=1
```

I believe it's enough for this case. Also you could use this 2 in a constant (i.e. `CONFIDENCE=2` and then `detection[CONFIDENCE]`)

Reply:
Done.